pdns/common_startup.cc

```diff
@@ -238,6 +238,7 @@ void declareArguments()
   ::arg().set("tcp-fast-open", "Enable TCP Fast Open support on the listening sockets, using the supplied numerical value as the queue size")="0";
 
   ::arg().set("max-generate-steps", "Maximum number of $GENERATE steps when loading a zone from a file")="0";
+  ::arg().setSwitch("upgrade-unknown-types","Transparently upgrade known TYPExxx records. Recommended to keep off, except for PowerDNS upgrades until data sources are cleaned up")="no";
```

Comment:
Are we sure about the "upgrade-unknown-types" because the comment says "upgrade known TYPExxx records" and this pull is upgrading known types.

Maybe "upgrade-known-typex-records" is a better name or "serve-known-typex-records". But my "naming things" skills are terrible, so feel free to pick something else.

Reply:
I had pretty much the same question myself. But I too suck at naming things.

IDK if "typex-records" is something understandable. Maybe "serve-hex-encoded-records"?

Reply:
> Are we sure about the "upgrade-unknown-types" because the comment says "upgrade known TYPExxx records" and this pull is upgrading known types.

It's upgrading the known unknown types. I think it might be fine just like this :)

Reply:
Okay, lets ship with this name for now.